updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -52,11 +52,16 @@ def perform
           Dependabot.logger.info("Starting update job for #{job.source.repo}")
           Dependabot.logger.info("Checking all dependencies for version updates...")
 
+          # Raise an error if the package manager version is unsupported
+          dependency_snapshot.package_manager&.raise_if_unsupported!
+
           # Retrieve the list of initial notices from dependency snapshot
           @notices = dependency_snapshot.notices
           # More notices can be added during the update process
 
           dependencies.each { |dep| check_and_create_pr_with_error_handling(dep) }
+        rescue StandardError => e
+          error_handler.handle_dependency_error(error: e, dependency: dependencies.last)
         end
```

Comment:
When we have `unsupported error` for the package manager, we don't want to go over each dependency, instead we want to raise unsupported error for the whole process and handle error properly.